tests/db/model/generator/withCollectors.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/db/model/generator/withCollectors.js
+ */
+'use strict';
+const expect = require('chai').expect;
+const tu = require('../../../testUtils');
+const u = require('./utils');
+const gtUtil = u.gtUtil;
+const Generator = tu.db.Generator;
+const GeneratorTemplate = tu.db.GeneratorTemplate;
+const ZERO = 0;
+const ONE = 1;
+const TWO = 2;
+
+describe('tests/db/model/generator/create.js >', () => {
+  let createdGenerator;
+  const generator = JSON.parse(JSON.stringify(u.getGenerator()));
+  const generatorTemplate = gtUtil.getGeneratorTemplate();
+  let userInst;
+  before((done) => {
+    tu.createUser('GeneratorOwner')
+    .then((user) => {
+      userInst = user;
+      generator.createdBy = user.id;
+      return GeneratorTemplate.create(generatorTemplate);
+    })
+    .then(() => Generator.create(generator))
+    .then((_createdGenerator) => {
+      createdGenerator = _createdGenerator;
+      done();
+    })
+    .catch(done);
+  });
+
+  after(u.forceDelete);
+  after(gtUtil.forceDelete);
+
+  describe('with one collector', () => {
+    let collector;
+    let relodedGenerator;
+
+    before((done) => {
+      tu.db.Collector.create({ name: 'snow' })
+      .then((_collector) => {
+        collector = _collector;
+        return createdGenerator.addCollectors([collector]);
```

Comment:
@iamigo @shriramshankar given that collectors are added onto generators, shall we still POST with a collector?
if yes, the collectors would be added onto generators in the db/models, and the validation for whether the collector exists and is registered would be in the generator controller.

Reply:
Updated PR with the changes you mentioned @shriramshankar @iamigo

Reply:
yes, we want to post a new generator with a list of collectors
if there is no match for any of the collectors specified in the list, then reject the POST
i don't care if the check is made in the api layer or the db layer, but if it's in the db layer it would need to be in a beforeCreate hook